Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -266,7 +266,9 @@ public class MapboxNavigationService: NSObject, NavigationService {
     
     deinit {
         suspendNotifications()
-        endNavigation()
+        eventsManager.withBackupDataSource(self) {
```

Comment:
Looks too hacky, but I can't really come to a better solution. Making `dataSource` strong seems to be even worse.

Reply:
Same story...